File: lcr/184/main.go
package main

type Checkout struct {
	Queue    []int
	MaxQueue []int
}

func Constructor() Checkout {
	return Checkout{
		Queue:    []int{},
		MaxQueue: []int{},
	}
}

func (this *Checkout) Get_max() int {
	if len(this.Queue) == 0 {
		return -1
	}
	return this.MaxQueue[0]
}

func (this *Checkout) Add(value int) {
	this.Queue = append(this.Queue, value)

	for len(this.MaxQueue) > 0 && value > this.MaxQueue[len(this.MaxQueue)-1] {
		this.MaxQueue = this.MaxQueue[:len(this.MaxQueue)-1]
	}
	this.MaxQueue = append(this.MaxQueue, value)
}

func (this *Checkout) Remove() int {
	if len(this.Queue) == 0 {
		return -1
	}
	value := this.Queue[0]
	this.Queue = this.Queue[1:]

	if len(this.MaxQueue) > 0 && value == this.MaxQueue[0] {
		this.MaxQueue = this.MaxQueue[1:]
	}

	return value
}

File: lcr/184/main.py
class MaxQueue:
    def __init__(self):
        self.queue = []
        self.max_queue = []

    def max_value(self) -> int:
        if not self.max_queue:
            return -1
        return self.max_queue[0]

    def push_back(self, value: int) -> None:
        self.queue.append(value)

        while self.max_queue and self.max_queue[-1] < value:
            self.max_queue = self.max_queue[:-1]
        self.max_queue.append(value)

    def pop_front(self) -> int:
        if not self.queue:
            return -1

        while self.max_queue and self.max_queue[0] == self.queue[0]:
            self.max_queue = self.max_queue[1:]

        value = self.queue[0]
        self.queue = self.queue[1:]
        return value